lib/Sema/CSDiagnostics.h

```diff
@@ -441,15 +457,26 @@ class SuperclassRequirementFailure final : public RequirementFailure {
 /// Call to `foo` is going to be diagnosed as missing `q:`
 /// and having extraneous `a:` labels, with appropriate fix-its added.
 class LabelingFailure final : public FailureDiagnostic {
-  ArrayRef<Identifier> CorrectLabels;
+  using ParamBinding = SmallVector<unsigned, 1>;
+
+  bool IsComplex;
+  ArgumentRelabeling Relabeling;
```

Comment:
I wonder whether it would make more sense to provide fixes directly to this diagnostic instead of copying all this data, WDYT?

Reply:
I previously thought it would be better to loosely combine CSDiagnostic and CSFix.

However, I think it's better that way.
It makes the code simpler, and it expresses well the idea of creating a message from multiple label errors.

I will do.